src/com/sun/jna/Structure.java

```diff
@@ -888,7 +953,19 @@ protected void writeField(StructField structField) {
      * guaranteed to be predictable.
      * @return ordered list of field names
      */
-    protected abstract List<String> getFieldOrder();
+    // TODO(idosu 28 Apr 2018): Maybe deprecate this method to let users know they should use @FieldOrder
+    protected List<String> getFieldOrder() {
+        List<String> fields = new LinkedList<String>();
+        for (Class<?> clazz = getClass(); clazz != Structure.class; clazz = clazz.getSuperclass()) {
```

Comment:
If you walk the inheritance structure here, you'll miss classes, that extend Structure, but don't use the annotation. So what I would do at this point:

- fetch the field order for the super class via a `super.getFieldOrder()` call
- extract the local order from the annotation and append it

This way you can mix the definitions.

Reply:
**not fixed**

I thought about it but I don't agree, because if a class overrides this method no sub class can call the default implementation i.e. getting with annotation
``` java
class Parent extends Structure {
    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("a");
    }
}

class Son extends C {
    @Override
    protected List<String> getFieldOrder() {
        // super.getFieldOrder() returns [ "a" ]
        // Structure.super.getFieldOrder() - this is not valid
    }
}
```

plus calling `getClass()` in `super.getFieldOrder()` will result in the same class as it would if you write it here and therefore will result in a `StackOverflowException`.

Could you show me an example perhaps?

Reply:
BTW I had another idea to check in `this.fieldOrder` if we already got fields for the super class and use them instead of iterating the inheritance structure. what do you think, does it worth the effort?

Reply:
You are right - subclasses can still call `super#getFieldOrder` and so build a complete list if a super class uses the `@FieldOrder` annotation. It will fail if a child class is using the annotation and the super class is not. In this case the member check at runtime will blow. For the 5.0.0 release it would be ok to update all structures in the platform.